src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgeryTokenSerializer.cs

```diff
@@ -156,7 +161,14 @@ public string Serialize(AntiforgeryToken token)
 
                 writer.Flush();
                 var stream = serializationContext.Stream;
-                return WebEncoders.Base64UrlEncode(_cryptoSystem.Protect(stream.ToArray()));
+                var bytes = _cryptoSystem.Protect(stream.ToArray());
```

Comment:
Because the `IDataProtector.Protect()` argument is a `byte[]` of the correct length and not an `ArraySegment<byte>`, @rynowak's approach would work only if we

1. Calculated the `MemoryStream` buffer size in advance.
2. Serialized a lot of tokens with _exactly_ the same initial serialization size.
3. Pooled / reused a few `byte[]` instances of different sizes.
4. Managed contention between requests.